SSL/TLS Exception with a website... https://latinlexicon.org/

```
13:48:36.732 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.h.Http1FrontendHandler - [Client (kubernetes.docker.internal:58258)] <=> [PROXY] : handlerAdded
13:48:36.737 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsBackendHandler - [Client (kubernetes.docker.internal:58258)] <=> [Server (latinlexicon.org:443)] : handlerAdded
13:48:36.739 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.h.Http1FrontendHandler - [Client (kubernetes.docker.internal:58258)] <=> [Server (latinlexicon.org:443)] : DefaultFullHttpResponse
13:48:36.739 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsFrontendHandler - [Client (kubernetes.docker.internal:58258)] <=> [Server (latinlexicon.org:443)] : handlerAdded
13:48:36.740 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsFrontendHandler - [Client (kubernetes.docker.internal:58258)] <=> [Server (latinlexicon.org:443)] : handlerRemoved
13:48:36.740 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.h.Http1FrontendHandler - [Client (kubernetes.docker.internal:58258)] <=> [Server (latinlexicon.org:443)] : handlerRemoved
13:48:36.746 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsFrontendHandler - SSL detection with true
13:48:36.746 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsFrontendHandler - Client SNI lookup with latinlexicon.org
13:48:36.746 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsFrontendHandler - Client ALPN lookup with [h2, http/1.1]
13:48:36.780 [nioEventLoopGroup-3-7] DEBUG i.n.h.s.ReferenceCountedOpenSslContext - verification of certificate failed
		      sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
		      at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:439)
		      at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
		      at java.base/sun.security.validator.Validator.validate(Validator.java:264)
		      at java.base/sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
		      at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:276)
		      at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:141)
		      at io.netty.handler.ssl.ReferenceCountedOpenSslClientContext$ExtendedTrustManagerVerifyCallback.verify(ReferenceCountedOpenSslClientContext.java:234)
		      at io.netty.handler.ssl.ReferenceCountedOpenSslContext$AbstractCertificateVerifier.verify(ReferenceCountedOpenSslContext.java:723)
		      at io.netty.internal.tcnative.CertificateVerifierTask.runTask(CertificateVerifierTask.java:36)
		      at io.netty.internal.tcnative.SSLTask.run(SSLTask.java:38)
		      at io.netty.handler.ssl.ReferenceCountedOpenSslEngine$3.run(ReferenceCountedOpenSslEngine.java:1447)
		      at io.netty.handler.ssl.SslHandler.runAllDelegatedTasks(SslHandler.java:1512)
		      at io.netty.handler.ssl.SslHandler.runDelegatedTasks(SslHandler.java:1526)
		      at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1390)
		      at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible(SslHandler.java:1245)
		      at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1282)
		      at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507)
		      at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446)
		      at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
		      at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
		      at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
		      at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
		      at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
		      at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
		      at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
		      at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
		      at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
		      at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719)
		      at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
		      at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
		      at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
		      at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
		      at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		      at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		      at java.base/java.lang.Thread.run(Thread.java:834)
		      Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
		      at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
		      at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
		      at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
		      at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
		      ... 34 common frames omitted
13:48:36.782 [nioEventLoopGroup-3-7] WARN  i.n.h.s.ApplicationProtocolNegotiationHandler - [id: 0x8d04bbdb, L:0.0.0.0/0.0.0.0:58217] TLS handshake failed:
    javax.net.ssl.SSLHandshakeException: General OpenSslEngine problem
    at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.handshakeException(ReferenceCountedOpenSslEngine.java:1845)
    at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.wrap(ReferenceCountedOpenSslEngine.java:812)
    at java.base/javax.net.ssl.SSLEngine.wrap(SSLEngine.java:522)
    at io.netty.handler.ssl.SslHandler.wrap(SslHandler.java:1039)
    at io.netty.handler.ssl.SslHandler.wrapNonAppData(SslHandler.java:925)
    at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1403)
    at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible(SslHandler.java:1245)
    at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1282)
    at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
    at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
    at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
    at java.base/java.lang.Thread.run(Thread.java:834)
    Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
    at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:439)
    at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
    at java.base/sun.security.validator.Validator.validate(Validator.java:264)
    at java.base/sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
    at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:276)
    at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:141)
    at io.netty.handler.ssl.ReferenceCountedOpenSslClientContext$ExtendedTrustManagerVerifyCallback.verify(ReferenceCountedOpenSslClientContext.java:234)
    at io.netty.handler.ssl.ReferenceCountedOpenSslContext$AbstractCertificateVerifier.verify(ReferenceCountedOpenSslContext.java:723)
    at io.netty.internal.tcnative.CertificateVerifierTask.runTask(CertificateVerifierTask.java:36)
    at io.netty.internal.tcnative.SSLTask.run(SSLTask.java:38)
    at io.netty.handler.ssl.ReferenceCountedOpenSslEngine$3.run(ReferenceCountedOpenSslEngine.java:1447)
    at io.netty.handler.ssl.SslHandler.runAllDelegatedTasks(SslHandler.java:1512)
    at io.netty.handler.ssl.SslHandler.runDelegatedTasks(SslHandler.java:1526)
    at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1390)
    ... 21 common frames omitted
    Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
    at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
    at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
    at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
    at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
    ... 34 common frames omitted
13:48:36.782 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsFrontendHandler - ALPN negotiate failed with javax.net.ssl.SSLHandshakeException: General OpenSslEngine problem
13:48:36.783 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsBackendHandler - [Client (kubernetes.docker.internal:58258)] <=> [Server (latinlexicon.org:443)] : channelInactive
13:48:36.783 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsBackendHandler - [Client (kubernetes.docker.internal:58258)] <=> [Server (latinlexicon.org:443)] : handlerRemoved
13:48:36.783 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsFrontendHandler - Client ALPN lookup with [h2, http/1.1]
13:48:36.784 [nioEventLoopGroup-3-7] DEBUG i.n.h.ssl.SslClientHelloHandler - Unexpected client hello packet: 1603010200010001fc0303a22f5a98894d617be0a8a7ffe1140e8c682faa34cc0f15ec57a52ec4c16bb891208d7a92d7215c817b0d0c42631aace8eddb7bc6f8d0c5dccd5550746d4ff56c2200204a4a130113021303c02bc02fc02cc030cca9cca8c013c014009c009d002f003501000193aaaa00000000001500130000106c6174696e6c657869636f6e2e6f726700170000ff01000100000a000a00081a1a001d00170018000b00020100002300000010000e000c02683208687474702f312e31000500050100000000000d0012001004030804040105030805050108060601001200000033002b00291a1a000100001d0020eb1ba601b28d9b1fa0a0dbc72cecad480abe324349c61190b6b42b775dd45579002d00020101002b000b0a7a7a0304030303020301001b00030200024469000500030268325a5a000100001500c3000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
    java.lang.IllegalStateException: complete already: DefaultPromise@11d850bb(success: [h2, http/1.1])
    at io.netty.util.concurrent.DefaultPromise.setSuccess(DefaultPromise.java:99)
    at com.github.chhsiao90.nitmproxy.handler.protocol.tls.TlsFrontendHandler$AlpnNegotiateHandler.lookup(TlsFrontendHandler.java:143)
			  at com.github.chhsiao90.nitmproxy.handler.protocol.tls.AbstractAlpnHandler.lookup(AbstractAlpnHandler.java:108)
			  at io.netty.handler.ssl.SslClientHelloHandler.select(SslClientHelloHandler.java:188)
			  at io.netty.handler.ssl.SslClientHelloHandler.decode(SslClientHelloHandler.java:149)
			  at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507)
			  at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446)
			  at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:404)
			  at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:371)
			  at io.netty.handler.codec.ByteToMessageDecoder.channelInactive(ByteToMessageDecoder.java:354)
			  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262)
			  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248)
			  at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:241)
			  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405)
			  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262)
			  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248)
			  at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901)
			  at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:831)
			  at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
			  at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
			  at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:497)
			  at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
			  at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
			  at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
			  at java.base/java.lang.Thread.run(Thread.java:834)
13:48:36.784 [nioEventLoopGroup-3-7] DEBUG c.g.c.n.h.p.tls.TlsFrontendHandler - Client ALPN lookup with null
13:48:36.785 [nioEventLoopGroup-3-7] WARN  i.n.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
			  io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: complete already: DefaultPromise@11d850bb(success: [h2, http/1.1])
			  at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:477)
			  at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:404)
			  at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:371)
			  at io.netty.handler.codec.ByteToMessageDecoder.channelInactive(ByteToMessageDecoder.java:354)
			  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262)
			  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248)
			  at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:241)
			  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405)
			  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262)
			  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248)
			  at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901)
			  at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:831)
			  at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
			  at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
			  at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:497)
			  at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		    at java.base/java.lang.Thread.run(Thread.java:834)
		    Caused by: java.lang.IllegalStateException: complete already: DefaultPromise@11d850bb(success: [h2, http/1.1])
		    at io.netty.util.concurrent.DefaultPromise.setSuccess(DefaultPromise.java:99)
		    at com.github.chhsiao90.nitmproxy.handler.protocol.tls.TlsFrontendHandler$AlpnNegotiateHandler.lookup(TlsFrontendHandler.java:143)
		    at com.github.chhsiao90.nitmproxy.handler.protocol.tls.AbstractAlpnHandler.lookup(AbstractAlpnHandler.java:108)
		    at io.netty.handler.ssl.SslClientHelloHandler.select(SslClientHelloHandler.java:188)
		    at io.netty.handler.ssl.SslClientHelloHandler.decode(SslClientHelloHandler.java:169)
		    at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507)
		    at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446)
		    ... 18 common frames omitted

```

--- a/src/main/java/com/github/chhsiao90/nitmproxy/NitmProxyConfig.java
+++ b/src/main/java/com/github/chhsiao90/nitmproxy/NitmProxyConfig.java
@@ -5,6 +5,7 @@ import com.github.chhsiao90.nitmproxy.handler.protocol.ProtocolDetector;
 import com.github.chhsiao90.nitmproxy.handler.protocol.http1.Http1ProtocolDetector;
 import com.github.chhsiao90.nitmproxy.listener.ForwardListener;
 import com.github.chhsiao90.nitmproxy.listener.HttpListener;
+import com.github.chhsiao90.nitmproxy.tls.UnsafeAccessSupport;
 import com.google.common.base.Joiner;
 import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
 import org.bouncycastle.cert.X509CertificateHolder;
@@ -44,7 +45,9 @@ public class NitmProxyConfig {
     private NitmProxyStatusListener statusListener;
     private List<HttpListener> httpListeners;
     private List<ForwardListener> forwardListeners;
+
     private TrustManager trustManager;
+    private UnsafeAccessSupport unsafeAccessSupport = UnsafeAccessSupport.DENY;
 
     private List<ProtocolDetector> detectors;
 
@@ -162,6 +165,14 @@ public class NitmProxyConfig {
         this.maxContentLength = maxContentLength;
     }
 
+    public NitmProxyStatusListener getStatusListener() {
+        return statusListener;
+    }
+
+    public void setStatusListener(NitmProxyStatusListener statusListener) {
+        this.statusListener = statusListener;
+    }
+
     public List<HttpListener> getHttpListeners() {
         return httpListeners;
     }
@@ -178,20 +189,20 @@ public class NitmProxyConfig {
         this.forwardListeners = forwardListeners;
     }
 
+    public UnsafeAccessSupport getUnsafeAccessSupport() {
+        return unsafeAccessSupport;
+    }
+
+    public void setUnsafeAccessSupport(UnsafeAccessSupport unsafeAccessSupport) {
+        this.unsafeAccessSupport = unsafeAccessSupport;
+    }
+
     public List<ProtocolDetector> getDetectors() {
         return detectors;
     }
 
     public void setDetectors(List<ProtocolDetector> detectors) {
         this.detectors = detectors;
-    }
-
-    public NitmProxyStatusListener getStatusListener() {
-        return statusListener;
-    }
-
-    public void setStatusListener(NitmProxyStatusListener statusListener) {
-        this.statusListener = statusListener;
     }
 
     @Override

--- a/src/main/java/com/github/chhsiao90/nitmproxy/tls/SimpleUnsafeAccessSupport.java
+++ b/src/main/java/com/github/chhsiao90/nitmproxy/tls/SimpleUnsafeAccessSupport.java
@@ -2,6 +2,7 @@ package com.github.chhsiao90.nitmproxy.tls;
 
 import com.github.chhsiao90.nitmproxy.Address;
 import com.github.chhsiao90.nitmproxy.ConnectionContext;
+import com.github.chhsiao90.nitmproxy.handler.protocol.http2.Http2FramesWrapper;
 import com.github.chhsiao90.nitmproxy.listener.HttpListener;
 import com.google.common.io.Resources;
 import io.netty.handler.codec.http.DefaultFullHttpResponse;
@@ -68,7 +69,7 @@ public class SimpleUnsafeAccessSupport implements UnsafeAccessSupport {
             }
             switch (accepted.get(connectionContext.getServerAddr())) {
                 case ASK:
-                    return handleAsk(connectionContext, request);
+                    return handleAskHttp1(connectionContext, request);
                 case DENY:
                     return Optional.of(createDenyResponse());
                 case ACCEPT:
@@ -77,17 +78,58 @@ public class SimpleUnsafeAccessSupport implements UnsafeAccessSupport {
             }
         }
 
-        private Optional<FullHttpResponse> handleAsk(ConnectionContext connectionContext, FullHttpRequest request) {
+        @Override
+        public Optional<Http2FramesWrapper> onHttp2Request(ConnectionContext context, Http2FramesWrapper request) {
+            if (context.getServerAddr() == null || !accepted.containsKey(context.getServerAddr())) {
+                return Optional.empty();
+            }
+            switch (accepted.get(context.getServerAddr())) {
+                case ASK:
+                    return handleAskHttp2(context, request);
+                case DENY:
+                    return Optional.of(Http2FramesWrapper
+                            .builder(request.getStreamId())
+                            .response(createDenyResponse())
+                            .build());
+                case ACCEPT:
+                default:
+                    return Optional.empty();
+            }
+        }
+
+        private Optional<FullHttpResponse> handleAskHttp1(ConnectionContext context, FullHttpRequest request) {
             if (request.uri().endsWith(ACCEPT_MAGIC)) {
                 request.setUri(request.uri().replace(ACCEPT_MAGIC, ""));
-                accepted.put(connectionContext.getServerAddr(), UnsafeAccess.ACCEPT);
+                accepted.put(context.getServerAddr(), UnsafeAccess.ACCEPT);
                 return Optional.empty();
             }
             if (request.uri().endsWith(DENY_MAGIC)) {
-                accepted.put(connectionContext.getServerAddr(), UnsafeAccess.DENY);
+                accepted.put(context.getServerAddr(), UnsafeAccess.DENY);
                 return Optional.of(createDenyResponse());
             }
             return Optional.of(createAskResponse(request.uri()));
+        }
+
+        private Optional<Http2FramesWrapper> handleAskHttp2(
+                ConnectionContext connectionContext,
+                Http2FramesWrapper request) {
+            String uri = request.getHeaders().headers().path().toString();
+            if (uri.endsWith(ACCEPT_MAGIC)) {
+                request.getHeaders().headers().path(uri.replace(ACCEPT_MAGIC, ""));
+                accepted.put(connectionContext.getServerAddr(), UnsafeAccess.ACCEPT);
+                return Optional.empty();
+            }
+            if (uri.endsWith(DENY_MAGIC)) {
+                accepted.put(connectionContext.getServerAddr(), UnsafeAccess.DENY);
+                return Optional.of(Http2FramesWrapper
+                        .builder(request.getStreamId())
+                        .response(createDenyResponse())
+                        .build());
+            }
+            return Optional.of(Http2FramesWrapper
+                    .builder(request.getStreamId())
+                    .response(createAskResponse(request.getHeaders().headers().path().toString()))
+                    .build());
         }
 
         private FullHttpResponse createDenyResponse() {

--- a/src/main/java/com/github/chhsiao90/nitmproxy/tls/SimpleUnsafeAccessSupport.java
+++ b/src/main/java/com/github/chhsiao90/nitmproxy/tls/SimpleUnsafeAccessSupport.java
@@ -1,0 +1,106 @@
+package com.github.chhsiao90.nitmproxy.tls;
+
+import com.github.chhsiao90.nitmproxy.Address;
+import com.github.chhsiao90.nitmproxy.ConnectionContext;
+import com.github.chhsiao90.nitmproxy.listener.HttpListener;
+import com.google.common.io.Resources;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+
+import javax.net.ssl.TrustManagerFactory;
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.google.common.io.Resources.*;
+import static io.netty.handler.codec.http.HttpHeaderNames.*;
+import static io.netty.handler.codec.http.HttpHeaderValues.*;
+import static java.lang.String.*;
+import static java.nio.charset.StandardCharsets.*;
+
+public class SimpleUnsafeAccessSupport implements UnsafeAccessSupport {
+
+    private static final String ACCEPT_MAGIC = ";nitmproxy-unsafe=accept";
+    private static final String DENY_MAGIC = ";nitmproxy-unsafe=deny";
+
+    private final ConcurrentMap<Address, UnsafeAccess> accepted;
+    private final String askTemplate;
+    private final Interceptor interceptor;
+
+    @SuppressWarnings("UnstableApiUsage")
+    public SimpleUnsafeAccessSupport() throws IOException {
+        this(Resources.toString(getResource("html-templates/ask-unsafe-access.html"), UTF_8));
+    }
+
+    public SimpleUnsafeAccessSupport(String askTemplate) {
+        this.accepted = new ConcurrentHashMap<>();
+        this.askTemplate = askTemplate;
+        this.interceptor = new Interceptor();
+    }
+
+    @Override
+    public UnsafeAccess checkUnsafeAccess(ConnectionContext context, X509Certificate[] chain,
+            CertificateException cause) {
+        accepted.putIfAbsent(context.getServerAddr(), UnsafeAccess.ASK);
+        return accepted.get(context.getServerAddr());
+    }
+
+    @Override
+    public TrustManagerFactory create(TrustManagerFactory delegate, ConnectionContext context) {
+        return UnsafeAccessSupportTrustManagerFactory.create(delegate, this, context);
+    }
+
+    public HttpListener getInterceptor() {
+        return interceptor;
+    }
+
+    class Interceptor implements HttpListener {
+        @Override
+        public Optional<FullHttpResponse> onHttp1Request(ConnectionContext connectionContext, FullHttpRequest request) {
+            if (connectionContext.getServerAddr() == null || !accepted.containsKey(connectionContext.getServerAddr())) {
+                return Optional.empty();
+            }
+            switch (accepted.get(connectionContext.getServerAddr())) {
+                case ASK:
+                    return handleAsk(connectionContext, request);
+                case DENY:
+                    return Optional.of(createDenyResponse());
+                case ACCEPT:
+                default:
+                    return Optional.empty();
+            }
+        }
+
+        private Optional<FullHttpResponse> handleAsk(ConnectionContext connectionContext, FullHttpRequest request) {
+            if (request.uri().endsWith(ACCEPT_MAGIC)) {
+                request.setUri(request.uri().replace(ACCEPT_MAGIC, ""));
+                accepted.put(connectionContext.getServerAddr(), UnsafeAccess.ACCEPT);
+                return Optional.empty();
+            }
+            if (request.uri().endsWith(DENY_MAGIC)) {
+                accepted.put(connectionContext.getServerAddr(), UnsafeAccess.DENY);
+                return Optional.of(createDenyResponse());
+            }
+            return Optional.of(createAskResponse(request.uri()));
+        }
+
+        private FullHttpResponse createDenyResponse() {
+            return new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.FORBIDDEN);
+        }
+
+        private FullHttpResponse createAskResponse(String uri) {
+            DefaultFullHttpResponse response =  new DefaultFullHttpResponse(
+                    HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+            response.content().writeCharSequence(format(askTemplate, uri + ACCEPT_MAGIC, uri + DENY_MAGIC), UTF_8);
+            response.headers().set(CONTENT_TYPE, TEXT_HTML);
+            response.headers().set(CONTENT_LENGTH, response.content().readableBytes());
+            return response;
+        }
+    }
+}

--- a/src/main/java/com/github/chhsiao90/nitmproxy/tls/TlsContext.java
+++ b/src/main/java/com/github/chhsiao90/nitmproxy/tls/TlsContext.java
@@ -10,6 +10,7 @@ import static java.util.Collections.*;
 public class TlsContext {
 
     private boolean enabled = true;
+    private UnsafeAccess unsafeAccess = UnsafeAccess.ACCEPT;
     private Promise<List<String>> protocols;
     private Promise<String> protocol;
 
@@ -18,6 +19,13 @@ public class TlsContext {
         return this;
     }
 
+    public void askUnsafeAccess() {
+        unsafeAccess = UnsafeAccess.ASK;
+    }
+
+    public UnsafeAccess unsafeAccess() {
+        return unsafeAccess;
+    }
     /**
      * Get the ALPN protocols sent from the client.
      *

--- a/src/main/java/com/github/chhsiao90/nitmproxy/tls/TlsUtil.java
+++ b/src/main/java/com/github/chhsiao90/nitmproxy/tls/TlsUtil.java
@@ -8,6 +8,7 @@ import io.netty.handler.ssl.ApplicationProtocolConfig.SelectorFailureBehavior;
 import io.netty.handler.ssl.SslContext;
 import io.netty.handler.ssl.SslContextBuilder;
 import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.netty.handler.ssl.util.TrustManagerFactoryWrapper;
 
 import javax.net.ssl.SSLException;
 import javax.net.ssl.TrustManagerFactory;
@@ -40,17 +41,10 @@ public final class TlsUtil {
                 .forClient()
                 .protocols(context.config().getTlsProtocols())
                 .sslContextProvider(context.config().getSslProvider())
-                .applicationProtocolConfig(applicationProtocolConfig(context.tlsCtx()));
+                .applicationProtocolConfig(applicationProtocolConfig(context.tlsCtx()))
+                .trustManager(trustManagerFactory(context));
         if (context.config().getClientKeyManagerFactory() != null) {
             builder.keyManager(context.config().getClientKeyManagerFactory());
-        }
-        if (context.config().getTrustManager() != null) {
-            builder.trustManager(context.config().getTrustManager());
-        }
-        if (context.config().isInsecure()) {
-            builder.trustManager(InsecureTrustManagerFactory.INSTANCE);
-        } else if (TRUST_MANAGER_FACTORY != null && context.config().getTrustManager() == null) {
-            builder.trustManager(TRUST_MANAGER_FACTORY);
         }
         return builder.build();
     }
@@ -84,5 +78,19 @@ public final class TlsUtil {
             }
         }
         return new String[] { HTTP_1_1 };
+    }
+
+    private static TrustManagerFactory trustManagerFactory(ConnectionContext context) {
+        UnsafeAccessSupport unsafeAccessSupport = context.config().getUnsafeAccessSupport();
+        if (context.config().getTrustManager() != null) {
+            return unsafeAccessSupport.create(
+                    new TrustManagerFactoryWrapper(context.config().getTrustManager()),
+                    context);
+        } else if (context.config().isInsecure()) {
+            return unsafeAccessSupport.create(InsecureTrustManagerFactory.INSTANCE, context);
+        } else if (TRUST_MANAGER_FACTORY != null) {
+            return unsafeAccessSupport.create(TRUST_MANAGER_FACTORY, context);
+        }
+        return null;
     }
 }

--- a/src/main/java/com/github/chhsiao90/nitmproxy/tls/UnsafeAccess.java
+++ b/src/main/java/com/github/chhsiao90/nitmproxy/tls/UnsafeAccess.java
@@ -1,0 +1,7 @@
+package com.github.chhsiao90.nitmproxy.tls;
+
+public enum UnsafeAccess {
+    ACCEPT,
+    DENY,
+    ASK
+}

--- a/src/main/java/com/github/chhsiao90/nitmproxy/tls/UnsafeAccessSupport.java
+++ b/src/main/java/com/github/chhsiao90/nitmproxy/tls/UnsafeAccessSupport.java
@@ -1,0 +1,27 @@
+package com.github.chhsiao90.nitmproxy.tls;
+
+import com.github.chhsiao90.nitmproxy.ConnectionContext;
+
+import javax.net.ssl.TrustManagerFactory;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public interface UnsafeAccessSupport {
+
+    UnsafeAccessSupport DENY = new UnsafeAccessSupport() {
+        @Override
+        public UnsafeAccess checkUnsafeAccess(ConnectionContext context, X509Certificate[] chain,
+                CertificateException cause) {
+            return UnsafeAccess.DENY;
+        }
+
+        @Override
+        public TrustManagerFactory create(TrustManagerFactory delegate, ConnectionContext context) {
+            return delegate;
+        }
+    };
+
+    UnsafeAccess checkUnsafeAccess(ConnectionContext context, X509Certificate[] chain, CertificateException cause);
+
+    TrustManagerFactory create(TrustManagerFactory delegate, ConnectionContext context);
+}

--- a/src/main/java/com/github/chhsiao90/nitmproxy/tls/UnsafeAccessSupportTrustManagerFactory.java
+++ b/src/main/java/com/github/chhsiao90/nitmproxy/tls/UnsafeAccessSupportTrustManagerFactory.java
@@ -1,0 +1,102 @@
+package com.github.chhsiao90.nitmproxy.tls;
+
+import com.github.chhsiao90.nitmproxy.ConnectionContext;
+import io.netty.handler.ssl.util.SimpleTrustManagerFactory;
+
+import javax.net.ssl.ManagerFactoryParameters;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import java.security.KeyStore;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+
+public class UnsafeAccessSupportTrustManagerFactory extends SimpleTrustManagerFactory {
+
+    private final TrustManager tm;
+
+    public UnsafeAccessSupportTrustManagerFactory(
+            X509TrustManager delegate,
+            UnsafeAccessSupport unsafeAccessSupport,
+            ConnectionContext context) {
+        this.tm = new UnsafeAccessSupportTrustManager(delegate, unsafeAccessSupport, context);
+    }
+
+    public static TrustManagerFactory create(
+            TrustManagerFactory factory,
+            UnsafeAccessSupport unsafeAccessSupport,
+            ConnectionContext context) {
+        return Arrays.stream(factory.getTrustManagers())
+                .filter(X509TrustManager.class::isInstance)
+                .map(X509TrustManager.class::cast)
+                .findFirst()
+                .<TrustManagerFactory>map(tm -> new UnsafeAccessSupportTrustManagerFactory(
+                        tm, unsafeAccessSupport, context))
+                .orElse(factory);
+    }
+
+    @Override
+    protected void engineInit(KeyStore keyStore) {
+    }
+
+    @Override
+    protected void engineInit(ManagerFactoryParameters managerFactoryParameters) {
+    }
+
+    @Override
+    protected TrustManager[] engineGetTrustManagers() {
+        return new TrustManager[] { tm };
+    }
+
+    private static class UnsafeAccessSupportTrustManager implements X509TrustManager {
+
+        private final X509TrustManager delegate;
+        private final UnsafeAccessSupport unsafeAccessSupport;
+        private final ConnectionContext context;
+
+        public UnsafeAccessSupportTrustManager(
+                X509TrustManager delegate,
+                UnsafeAccessSupport unsafeAccessSupport,
+                ConnectionContext context) {
+            this.delegate = delegate;
+            this.unsafeAccessSupport = unsafeAccessSupport;
+            this.context = context;
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+            try {
+                delegate.checkClientTrusted(chain, authType);
+            } catch (CertificateException e) {
+                switch (unsafeAccessSupport.checkUnsafeAccess(context, chain, e)) {
+                    case ACCEPT:
+                        break;
+                    case DENY:
+                        throw e;
+                    case ASK:
+                        context.tlsCtx().askUnsafeAccess();
+                }
+            }
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+            try {
+                delegate.checkServerTrusted(chain, authType);
+            } catch (CertificateException e) {
+                switch (unsafeAccessSupport.checkUnsafeAccess(context, chain, e)) {
+                    case ACCEPT:
+                        break;
+                    case DENY:
+                        throw e;
+                }
+            }
+        }
+
+        @Override
+        public X509Certificate[] getAcceptedIssuers() {
+            return delegate.getAcceptedIssuers();
+        }
+    }
+}

--- a/src/main/resources/html-templates/ask-unsafe-access.html
+++ b/src/main/resources/html-templates/ask-unsafe-access.html
@@ -1,0 +1,6 @@
+<html>
+<body>
+  <a href="%s">Accept</a>
+  <a href="%s">Reject</a>
+</body>
+</html>

--- a/src/test/java/com/github/chhsiao90/nitmproxy/testing/ByteBufAssert.java
+++ b/src/test/java/com/github/chhsiao90/nitmproxy/testing/ByteBufAssert.java
@@ -1,0 +1,27 @@
+package com.github.chhsiao90.nitmproxy.testing;
+
+import io.netty.buffer.ByteBuf;
+import org.assertj.core.api.AbstractObjectAssert;
+import org.assertj.core.api.InstanceOfAssertFactory;
+
+import static java.nio.charset.StandardCharsets.*;
+import static org.junit.Assert.*;
+
+public class ByteBufAssert extends AbstractObjectAssert<ByteBufAssert, ByteBuf> {
+    public ByteBufAssert(ByteBuf actual) {
+        super(actual, ByteBufAssert.class);
+    }
+
+    public static InstanceOfAssertFactory<ByteBuf, ByteBufAssert> asByteBuf() {
+        return new InstanceOfAssertFactory<>(ByteBuf.class, ByteBufAssert::new);
+    }
+
+    public ByteBufAssert hasContent(String expect) {
+        assertEquals(expect, actual.toString(UTF_8));
+        return this;
+    }
+
+    public void release() {
+        actual.release();
+    }
+}

--- a/src/test/java/com/github/chhsiao90/nitmproxy/testing/ResponseAssert.java
+++ b/src/test/java/com/github/chhsiao90/nitmproxy/testing/ResponseAssert.java
@@ -1,20 +1,22 @@
 package com.github.chhsiao90.nitmproxy.testing;
 
-import static io.netty.handler.codec.http.HttpResponseStatus.OK;
-import static java.nio.charset.StandardCharsets.UTF_8;
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.Assert.assertEquals;
-
-import org.assertj.core.api.AbstractObjectAssert;
-import org.assertj.core.api.StringAssert;
-
 import io.netty.handler.codec.http.FullHttpResponse;
 import io.netty.handler.codec.http.HttpResponseStatus;
+import org.assertj.core.api.AbstractObjectAssert;
+import org.assertj.core.api.InstanceOfAssertFactory;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.*;
 
 public class ResponseAssert extends AbstractObjectAssert<ResponseAssert, FullHttpResponse> {
 
   public ResponseAssert(FullHttpResponse actual) {
     super(actual, ResponseAssert.class);
+  }
+
+  public static InstanceOfAssertFactory<FullHttpResponse, ResponseAssert> asResponse() {
+    return new InstanceOfAssertFactory<>(FullHttpResponse.class, ResponseAssert::new);
   }
 
   public ResponseAssert ok() {
@@ -27,8 +29,8 @@ public class ResponseAssert extends AbstractObjectAssert<ResponseAssert, FullHtt
     return this;
   }
 
-  public StringAssert text() {
-    return new StringAssert(actual.content().toString(UTF_8));
+  public ByteBufAssert content() {
+    return new ByteBufAssert(actual.content());
   }
 
   public ResponseAssert hasHeader(String name, String value) {

--- a/src/test/java/com/github/chhsiao90/nitmproxy/tls/SimpleUnsafeAccessSupportTest.java
+++ b/src/test/java/com/github/chhsiao90/nitmproxy/tls/SimpleUnsafeAccessSupportTest.java
@@ -1,0 +1,103 @@
+package com.github.chhsiao90.nitmproxy.tls;
+
+import com.github.chhsiao90.nitmproxy.Address;
+import com.github.chhsiao90.nitmproxy.ConnectionContext;
+import com.github.chhsiao90.nitmproxy.tls.SimpleUnsafeAccessSupport.Interceptor;
+import io.netty.handler.codec.http.FullHttpRequest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import static com.github.chhsiao90.nitmproxy.http.HttpUtil.*;
+import static com.github.chhsiao90.nitmproxy.testing.ResponseAssert.*;
+import static com.github.chhsiao90.nitmproxy.tls.UnsafeAccess.*;
+import static io.netty.handler.codec.http.HttpMethod.*;
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+import static io.netty.handler.codec.http.HttpVersion.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+public class SimpleUnsafeAccessSupportTest {
+    private SimpleUnsafeAccessSupport unsafeAccessSupport;
+    private Interceptor interceptor;
+    private ConnectionContext context;
+    private FullHttpRequest request;
+
+    @Before
+    public void setUp() throws IOException {
+        context = mock(ConnectionContext.class);
+        when(context.getServerAddr()).thenReturn(new Address("www.google.com", 443));
+
+        unsafeAccessSupport = new SimpleUnsafeAccessSupport();
+        interceptor = (Interceptor) unsafeAccessSupport.getInterceptor();
+        request = request(HTTP_1_1, GET, "www.google.com", "/");
+    }
+
+    @After
+    public void tearDown() {
+        request.release();
+    }
+
+    @Test
+    public void shouldGetAsk() {
+        assertEquals(ASK, unsafeAccessSupport.checkUnsafeAccess(context, new X509Certificate[0], new CertificateException()));
+    }
+
+    @Test
+    public void shouldNotIntercept() {
+        assertThat(interceptor.onHttp1Request(context, request)).isEmpty();
+    }
+
+    @Test
+    public void shouldInterceptOnAsk() {
+        assertEquals(ASK, unsafeAccessSupport.checkUnsafeAccess(context, new X509Certificate[0], new CertificateException()));
+
+        assertThat(interceptor.onHttp1Request(context, request))
+                .isPresent()
+                .get(asResponse())
+                .content()
+                .hasContent("<html>\n" +
+                            "<body>\n" +
+                            "  <a href=\"/;nitmproxy-unsafe=accept\">Accept</a>\n" +
+                            "  <a href=\"/;nitmproxy-unsafe=deny\">Reject</a>\n" +
+                            "</body>\n" +
+                            "</html>")
+                .release();
+    }
+
+    @Test
+    public void shouldInterceptOnAccept() {
+        assertEquals(ASK, unsafeAccessSupport.checkUnsafeAccess(context, new X509Certificate[0], new CertificateException()));
+
+        request.setUri("/;nitmproxy-unsafe=accept");
+        assertThat(interceptor.onHttp1Request(context, request)).isEmpty();
+        assertEquals("/", request.uri());
+
+        assertEquals(ACCEPT, unsafeAccessSupport.checkUnsafeAccess(context, new X509Certificate[0], new CertificateException()));
+    }
+
+    @Test
+    public void shouldInterceptOnDeny() {
+        assertEquals(ASK, unsafeAccessSupport.checkUnsafeAccess(context, new X509Certificate[0], new CertificateException()));
+
+        request.setUri("/;nitmproxy-unsafe=deny");
+        assertThat(interceptor.onHttp1Request(context, request))
+                .isPresent()
+                .get(asResponse())
+                .status(FORBIDDEN)
+                .release();
+        assertEquals(DENY, unsafeAccessSupport.checkUnsafeAccess(context, new X509Certificate[0], new CertificateException()));
+
+        request.setUri("/");
+        assertThat(interceptor.onHttp1Request(context, request))
+                .isPresent()
+                .get(asResponse())
+                .status(FORBIDDEN)
+                .release();
+    }
+}

